templates/Dockerfile

```
@@ -68,7 +68,8 @@ ARG BUILDARCH
 RUN wget -O /bin/imgpkg https://github.com/vmware-tanzu/carvel-imgpkg/releases/download/${IMGPKG_VERSION}/imgpkg-linux-${BUILDARCH} && \
     chmod +x /bin/imgpkg
 
-# Install Tanzu CLI Plugin Builder
+# Install Tanzu CLI Plugin Builder.
+# Note: We are temporarily using a deactivated plugin until a bug fix is made to the plugin builder.
 FROM base AS cli-plugin-builder-install
```

```suggestion
FROM --platform=${BUILDPLATFORM} $BUILDER_BASE_IMAGE AS cli-plugin-builder-install
```

so that this stage won't get invalidated whenever there is a change in the project code

---

Ah, so that's what `--platform=${BUILDPLATFORM} $BUILDER_BASE_IMAGE` is for. I always thought it was because we wanted to be explicit about platform and base image. Didn't realize it was for cache. Good to know, thanks!

---

no, it's not for caching, what i'm saying is instead of creating the `cli-plugin-builder-install` build stage out of `base` stage which has the code baked in it, we can create one directly from `BUILDER_BASE_IMAGE` that way it is clean and it doesn't get invalidated whenever there is a change in the project code

An example for the above would be `test-base` stage https://github.com/vmware-tanzu/build-tooling-for-integrations/blob/main/templates/Dockerfile#L40 where we install envtest related dependencies and mount them to `test` stage so that whenever there is a change in code `test-base` doesn't get invalidated

---

Gotcha, thanks for the explanation.

I have made the change in a new commit.